httpx/client.py

```diff
@@ -277,13 +277,17 @@ def _get_auth_middleware(
             )
 
         if trust_env:
-            netrc_login = get_netrc_login(request.url.authority)
+            netrc_login = self._get_netrc_login(request.url.authority)
             if netrc_login:
                 username, _, password = netrc_login
                 return BasicAuthMiddleware(username=username, password=password)
 
         return None
 
+    @functools.lru_cache(1)
```

Comment:
Oh neat, I'd not considered that just applying it against a method like this.

One thing to consider - we *might* want to cache `get_netrc` rather than `get_netrc_login`, so that we're only looking up the netrc info once, rather than once per host.